blocks/table-of-contents/src/script.js

```diff
@@ -169,10 +173,24 @@ function renderList( children ) {
  * @param {HTMLElement} child
  * @return {HTMLElement} listItem
  */
-function renderListItem( child ) {
-	const listItem = document.createElement( 'li' );
-	listItem.appendChild( child );
-	return listItem;
+function renderListItem( regexAppend ) {
+	return (child) => {
+		const listItem = document.createElement( 'li' );
+		let appendText = '';
+		
+		if( isNonEmptyString( regexAppend ) ) {
+			appendText = (new RegExp( regexAppend, 'mi' )).exec( child.textContent );
+		}
```

Comment:
Can this be a function like `getAppendText( regexAppend, child.textContent)` so it can be assigned to const instead of let?